1045 grid editing remaining

jdilag90 opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**
Added remaining features from ticket #1045 
* add multiple rows at specific index
* add support for datepicker editor
* add support for timepicker editor
* addNewAtEnd setting to auto create/append rows in editable mode

**Related github/jira issue (required)**:
Closes #1045 

**Steps necessary to review your pull request (required)**:
1. Checkout branch, `npm install`, `npm run start`
2. Go to http://localhost:4300/ids-data-grid/editable.html

**Test Datepicker Editor**
* Click on a publishDate cell
* Date value should be autoselected
* Try selecting dates from date picker popup
* Cell should have dirty indicator if value changed

**Test Timepicker Editor**
* Click on a publishTime cell
* Time value should be autoselected
* Try selected times from time picker popup
* Cell should have dirty indicator if value changed

**Test addRows()**
* Open dev  console and run ```$('ids-data-grid').addRows([{}, {}, {}], 2)```
* It should insert 3 empty rows at row index 2
* Try different variations of rowData/rowIndex values

**Test addNewAtEnd**
1. Click on an editable cell on the last row
2. Hit Tab until a new row is appended to the data grid
3. Editable cells should be activated on focus

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

tmcconechy commented on doc/CHANGELOG.md:
```diff
@@ -28,6 +28,10 @@
 - `[DataGrid]` Added support for context menu. ([#963](https://github.com/infor-design/enterprise-wc/issues/963))
 - `[DataGrid]` Added support for editing. ([#991](https://github.com/infor-design/enterprise-wc/issues/991))
 - `[DataGrid]` Added support for dropdown editing ([#1045](https://github.com/infor-design/enterprise-wc/issues/1045))
+- `[DataGrid]` Added support for timepicker editing ([#1045](https://github.com/infor-design/enterprise-wc/issues/1045))
```
Change log should be in the beta 4 section right now...

tmcconechy reviewed (changes requested):
Looks great just a few suggestions.

1. when you click into the cell, you can see a 1 or 2 px shift of the icon - this is a bit worse in small row height (so do try all the row heights)
2. when clicking the trigger icon i think it should open it immediately similar to the dropdown (but only clicking the icon). At the moment it takes two clicks
3.  testing row heights i think the icon should have the same distance on the right as there is for the text on the left...
![Screenshot 2023-02-02 at 10 32 14 AM](https://user-images.githubusercontent.com/814283/216368693-85e50e81-0f89-4fa8-ae1a-d1630c9c3bf4.png)

 Nice work and attention to detail. thats all i can see so far

jdilag90 commented:
Addressed pr comments
* time picker and date picker popups auto open if icon clicked when hovering
* fixed hover and date/time picker icon alignment
* ids-trigger-field field-height spacing has another ticket (https://github.com/infor-design/enterprise-wc/issues/1091)

EdwardCoyle reviewed (commented):
This looks great!  The only issue I found is related to focusing the timepicker field -- It seems to cause some sort of issue with the grid's scrollbars.  Maybe something related to display type of the popup?  I was also still seeing the icon misalignment only in the Timepicker field (all sizes):

[Untitled_ Feb 6 2023 10_22 AM.webm](https://user-images.githubusercontent.com/3249601/217011778-685edba0-1a75-4b94-baba-46d233d92a3c.webm)

tmcconechy commented:
@EdwardCoyle we might need https://github.com/infor-design/enterprise-wc/issues/1064 -> i think all the issues of things being cut off need to be fixed. So told @jdilag to skip any of all that until these are all resolved.

tmcconechy commented:
@EdwardCoyle i took a look and its definitely ids-time-picker-popup 's placement. Do we have more fixes planned for that? Or maybe we make a new issue as i cant see an easy way while the popup is inside the grid cell. Think it needs to be outside.

So punt for now but do we make a new issue or is it part of another?

EdwardCoyle commented:
#1064 will cover the placement problems so we can leave it there
